auth/callback.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/absmach/supermq/pkg/errors"
+	svcerr "github.com/absmach/supermq/pkg/errors/service"
+	"github.com/absmach/supermq/pkg/policies"
+	"golang.org/x/sync/errgroup"
+)
+
+type callback struct {
+	httpClient *http.Client
+	urls       []string
+	method     string
+}
+
+// CallBack send auth request to an external service.
+//
+//go:generate mockery --name CallBack --output=./mocks --filename callback.go --quiet --note "Copyright (c) Abstract Machines"
+type CallBack interface {
+	Authorize(ctx context.Context, pr policies.Policy) error
+}
+
+// NewCallback creates a new instance of CallBack.
+func NewCallback(httpClient *http.Client, method string, urls []string) CallBack {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+
+	return &callback{
+		httpClient: httpClient,
+		urls:       urls,
+		method:     method,
+	}
+}
+
+func (c *callback) Authorize(ctx context.Context, pr policies.Policy) error {
+	payload := map[string]string{
+		"domain":           pr.Domain,
+		"subject":          pr.Subject,
+		"subject_type":     pr.SubjectType,
+		"subject_kind":     pr.SubjectKind,
+		"subject_relation": pr.SubjectRelation,
+		"object":           pr.Object,
+		"object_type":      pr.ObjectType,
+		"object_kind":      pr.ObjectKind,
+		"relation":         pr.Relation,
+		"permission":       pr.Permission,
+	}
+
+	if len(c.urls) == 0 {
+		return nil
+	}
```

Comment:
Move this to the beginning of the function. Also, let's use a single URL at a time, with other URLs being a fallback, something like [https://www.rabbitmq.com/docs/access-control#combined-backends](https://www.rabbitmq.com/docs/access-control#combined-backends).